charts/argocd-notifications/templates/service-metrics.yaml

```
@@ -5,6 +5,8 @@ metadata:
   name: {{ include "argocd-notifications.name" . }}-metrics
   labels:
     {{- include "argocd-notifications.metrics.labels" . | nindent 4 }}
```

Also here, do not extend the template function as these labels are also set on the ServiceMonitor object, not only on the `Service` as a user would suppose.
```suggestion
    {{- include "argocd-notifications.metrics.labels" . | nindent 4 }}
    {{- with .Values.metrics.service.labels }}
      {{- toYaml . | nindent 4 }}
    {{- end }}
```